### PR TITLE
Handle entities without IDs

### PR DESCRIPTION
**Pull request by seadowg**
- **Add failing test for null ID entities**
- **Don't return entities without an ID when updating**
- **Fix null ID on create case**
- **Allow clients to handle null entity IDs**

Closes #

### CODE REVIEW

File: src/test/java/org/javarosa/entities/EntitiesTest.java [hunk 1]
```diff
@@ -213,6 +213,39 @@ public void fillingFormWithUpdate_andNoLabel_makesEntityAvailableWithNullLabel()
         assertThat(entities.get(0).properties, equalTo(asList(new Pair<>("name", "Tom Wambsgans"))));
     }
 
+    @Test
+    public void fillingFormWithUpdate_withNullId_doesNotMakeEntityAvailable() throws IOException, XFormParser.ParseException {
+        Scenario scenario = Scenario.init("Create entity form", XFormsElement.html(
```
Comment (lognaturel) [inline src/test/java/org/javarosa/entities/EntitiesTest.java:259]: Title should be update not create

**Review by lognaturel — APPROVED** [review]
Minor naming issue and then this looks ready to merge to me. I’d prefer to squash because the direction changed but will leave that up to you.
